### DISCUSSION TITLE
能否开一讲WPF内存泄露的专题

### DISCUSSION THREAD

**Discussion by lanni1981** [BYJRK/DotNet-Discussions#28] [WPF 开发]
我做了一个截屏的WPF类库，每次调用以后，内存都在增加没有回收，能不能讲一讲内存泄露的问题。整个程序全部也就2M多一点，启动后就内存就占了70M，截屏一次增加10M，已经尝试了DeleteObject()、 GC.Collect()等办法。是不是wpf开发的软件都挺臃肿的？

**Comment by BYJRK** [comment]
这个你可能需要profiling一下，看一看到底是什么在占用内存。我只是听你说，也想不出来原因。我大概脑补了一下，每次截图都会在内存里保存一个Bitmap，导致占用了大量内存，而且你还没有显式Dispose掉，或者在某处留有引用，导致无法被GC回收？
________________________________
发件人: lanni1981 ***@***.***>
发送时间: 2023年2月16日 11:16
收件人: BYJRK/DotNet-Discussions ***@***.***>
抄送: Subscribed ***@***.***>
主题: [BYJRK/DotNet-Discussions] 能否开一讲WPF内存泄露的专题 (Discussion #28)


我做了一个截屏的WPF类库，每次调用以后，内存都在增加没有回收，能不能讲一讲内存泄露的问题。整个程序全部也就2M多一点，启动后就内存就占了70M，截屏一次增加10M，已经尝试了DeleteObject()、 GC.Collect()等办法。是不是wpf开发的软件都挺臃肿的？

―

**Comment by Hakoyu** [comment]
截屏? 我猜你没有释放流

**Comment by boogaloo-jw** [comment]
这样的问题，你需要在程序跑一段时间，抓个dump，然后用windbg工具进行分析，可以明确找到内存泄漏源！不用怀疑WPF，一定是你代码没写好😄

**Comment by lanni1981** [comment]
我不是程序员，就是爱好者，没能力分析。有能力的大佬帮我[看看](https://github.com/lanni1981/ScreenshotTest)
